lib/v2/zagg/new-arrivals.js

```diff
@@ -0,0 +1,46 @@
+const got = require('@/utils/got');
+const { art } = require('@/utils/render');
+const path = require('path');
+const cheerio = require('cheerio');
+const host = 'https://www.zagg.com/en_us';
+module.exports = async (ctx) => {
+    const { categories, brands } = ctx.params;
+
+    const url = `${host}/new-arrivals`;
+    const response = await got({
+        method: 'get',
+        url,
+        searchParams: {
+            cat: categories,
+            band: brands,
+        },
+    });
+    const data = response.data;
+
+    const $ = cheerio.load(data);
+    const list = $('.item.product.product-item')
+        .map(function () {
+            const data = {};
+            const details = $(this).find('.product.details-box').html();
+            data.link = $(this).find('.product-item-link').eq(0).attr('href');
+            data.title = $(this).find('.product-item-link').text();
+            const regex = /(https.*?)\?/;
+            const imgUrl = $(this).find('img').eq(0).attr('data-src').match(regex)[1];
+            const img = art(path.join(__dirname, 'templates/new-arrivals.art'), {
+                imgUrl,
+            });
+            data.description = details + img;
+            return data;
+        })
+        .get();
```

Comment:
u mean the web api for each product?

Reply:
POST to `url` with/without search parameters

Reply:
The same URL in 
https://github.com/DIYgod/RSSHub/blob/a9775c5d865f7b48ab781d5cf6976ef866a546e4/lib/v2/zagg/new-arrivals.js#L10-L17
It returns a JSON when you POST instead of GET

Reply:
on my side, when I GET the url , it returns 302 and redirects to the original website.

Reply:
could you please share the JSON file it returns?

Reply:
Set the header X-Requested-With to XMLHttpRequest and it will no longer 302

Reply:
awesome! let me try it!